contracts/main/stablecoin-core/StableSwapModule.sol

```diff
@@ -111,62 +98,61 @@ contract StableSwapModule is PausableUpgradeable, ReentrancyGuardUpgradeable, IS
         emit LogSetFeeOut(msg.sender, _feeOut);
     }
 
-    
-    function swapTokenToStablecoin(address _usr,uint256 _tokenAmount) external override nonReentrant whenNotPaused {
-        require(_tokenAmount != 0, "StableSwapModule/amount-zero");
-        
-        uint256 _tokenAmount18 = mul(_tokenAmount, to18ConversionFactor);
-        uint256 _fee = mul(_tokenAmount18, feeIn) / WAD;
-        uint256 _stablecoinAmount = sub(_tokenAmount18, _fee);
-        
-        require(tokenBalance[address(stablecoin)] >= _stablecoinAmount, "swapTokenToStablecoin/not-enough-stablecoin-balance");
-        
-        _udpateAndCheckDailyLimit(_tokenAmount);
-        
-        tokenBalance[address(stablecoin)] = sub(tokenBalance[address(stablecoin)], _stablecoinAmount);
-        tokenBalance[address(token)] = add(tokenBalance[address(token)], _tokenAmount);
-        totalFeeBalance += _fee;
-        
-        token.safeTransferFrom(msg.sender, address(this), _tokenAmount);
-        address(stablecoin).safeTransfer(_usr, _stablecoinAmount);
-        emit LogSwapTokenToStablecoin(_usr, _tokenAmount, _fee);
+    function swapTokenToStablecoin(address _usr, uint256 _amount) external override nonReentrant whenNotPaused {
+        require(_amount != 0, "StableSwapModule/amount-zero");
+
+        uint256 tokenAmount18 = _convertDecimals(_amount, IToken(token).decimals(), 18);
+        uint256 fee = (tokenAmount18 * feeIn) / WAD;
+        uint256 stablecoinAmount = tokenAmount18 - fee;
+        require(tokenBalance[stablecoin] >= stablecoinAmount, "swapTokenToStablecoin/not-enough-stablecoin-balance");
+
+        _udpateAndCheckDailyLimit(tokenAmount18);
+
+        tokenBalance[stablecoin] -= stablecoinAmount;
+        tokenBalance[token] += _amount;
+        totalFeeBalance += fee;
+
+        token.safeTransferFrom(msg.sender, address(this), _amount);
```

Comment:
_amount param is amount of token from which swap happens.
_amount param is amount of token that user pays